onadata/apps/main/forms.py

```diff
@@ -289,15 +295,16 @@ def publish(self, user, id_string=None, created_by=None):
                     default_storage.save(
                         upload_to(None, rand_name, user.username),
                         ContentFile(csv_data))
-            else:
+            if 'xls_file' in self.cleaned_data and\
```

Comment:
if `xls_file in self.cleaned_data` is true, isn't `self.cleaned_data['xls_file']` also true? can't we remove the second part